src/lib/s2k.c

```diff
@@ -85,7 +85,7 @@ pgp_s2k_round_iterations(size_t iterations)
 uint8_t
 pgp_s2k_encode_iterations(size_t iterations)
 {
-    for (uint8_t c = 0; c <= 255; ++c) {
+    for (uint8_t c = 0; c < 255; ++c) {
```

Comment:
Are you sure? Because I found one more implementation with old logic: https://github.com/golang/crypto/blob/master/openpgp/s2k/s2k.go#L75

Meawhile, RFC 4880, Section 3.7.2.3:
> For compatibility, when an S2K specifier is used, the special value
> 254 or 255 is stored in the position where the hash algorithm octet
> would have been in the old data structure.  This is then followed
> immediately by a one-octet algorithm identifier, and then by the S2K
> specifier as encoded above.

Looks like both possible but I think that old way is compabilited with another implementation.